ArduPlane/tiltrotor.cpp

```diff
@@ -592,6 +592,10 @@ void Tiltrotor::vectoring(void)
             motors->limit.yaw = true;
         }
 
+        // don't go beyond the configured yaw range
+        left_tilt = constrain_float(left_tilt, base_output - yaw_range, base_output + yaw_range);
+        right_tilt = constrain_float(right_tilt, base_output - yaw_range, base_output + yaw_range);
+
```

Comment:
I don't think this is in the correct place, it should constrain `tilt_offset` on 584, it also needs to set the yaw limit flag in that case.

Something like:
```
float tilt_scale = throttle_scaler * yaw_out * cos_tilt + avg_roll_factor * roll_out * sin_tilt;
if (fabsf(tilt_scale) > 1.0) {
    motors->limit.yaw = true;
    tilt_scale = constrain_float(tilt_scale, -1.0, 1.0);
}
const float tilt_offset = tilt_scale * yaw_range;
```

Reply:
it does set the limit.yaw flag, see line 592

Reply:
if we did it as you've proposed I think it wouldn't take into account the asymmetry properly. base_output is not 0.5

Reply:
If you add a new constrain you need to set the yaw limit flag if its its exceed. The current limit check is related to the constrains on 600 and 601.

568 and 587 do the same math as your constrain, your using `yaw_range` in place of `tilt_offset`. So we can just constrain `tilt_offset` directly. 

I can do a PR for my proposed fix if that would make the comparison easier.

Reply:
I see it now, fixed, thanks